conda-store-server/conda_store_server/action/generate_constructor_installer.py

```diff
@@ -20,13 +21,20 @@ def get_installer_platform():
     return context.subdir
 
 
+# TODO: Ideally, this would just use the lockfile instead of reading data
+# from the specification. But we currently also need to support legacy lockfiles
+# generated by conda-store, which were not generated by conda-lock, and those
+# don't have enough information for constructor to generate an installer. In
```

Comment:
I don't think this is true. You can pass the `explicit-list.txt` file (they contain an `@EXPLICIT` header) to [`environment_file`](https://conda.github.io/constructor/construct-yaml/#environment-file) and it should work. This is a relatively new feature introduced last year so it might need some polishing, but it's [part of the test suite](https://github.com/conda/constructor/blob/28309ac0d03093101de222d445c3150c1498ba90/tests/test_examples.py#L530-L541).

Reply:
Resolved. I've looked more into this. Turns out supporting explicit-lockfiles is not necessary here. First, they don't include pip dependencies, so the installer would miss those. Additionally, environments generating constructor artifacts will never have explicit-lockfiles, because builds using constructor would use new-style env-lockfiles, as env-style is the new default method (see `schema.BuildArtifactType.LOCKFILE` in `build_conda_environment`).

So I've changed the code that when a conda-store spec is provided, it tries to look up the lockfile and use that. There's a try/catch because it's in the external storage and is generated as part of a different task, but it works for me locally. On an exception, it just falls back to using the specification (non-pinned).